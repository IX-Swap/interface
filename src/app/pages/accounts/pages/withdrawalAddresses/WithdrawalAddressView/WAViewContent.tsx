import React from 'react'
import { Grid, LinearProgress } from '@material-ui/core'
import { LabelledValue } from 'components/LabelledValue'
import { useWithdrawalAddressById } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useWithdrawalAddressById'
import { useParams } from 'react-router-dom'

export const WAViewContent = () => {
  const params = useParams<{ withdrawalAddressId: string }>()
  const { data, isSuccess } = useWithdrawalAddressById(
    params.withdrawalAddressId
  )

  if (!isSuccess || data === undefined) {
    return <LinearProgress />
  }

  return (
    <>
      <Grid container direction='column' justify='center' spacing={3}>
        <Grid item xs={8}>
          <LabelledValue label='Blockchain Network' value={data.network.name} />
        </Grid>
        <Grid item xs={8}>
          <LabelledValue label='Address Label' value={data.label} />
        </Grid>
        <Grid item xs={8}>
          <LabelledValue label='Blockchain Address' value={data.address} />
        </Grid>
        <Grid item xs={8}>
          <LabelledValue label='Memo' value={data.memo} />
        </Grid>
      </Grid>
    </>
  )
}
