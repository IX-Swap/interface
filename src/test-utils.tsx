import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
