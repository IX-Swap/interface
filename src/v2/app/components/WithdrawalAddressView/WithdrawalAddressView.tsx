import React from 'react'
import { Grid, Box, Typography } from '@material-ui/core'
import { LabelledValue } from 'v2/components/LabelledValue'
import { useSetPageTitle } from 'v2/app/hooks/useSetPageTitle'
import { WithdrawalAddress } from 'v2/types/withdrawalAddress'
import CheckIcon from '@material-ui/icons/Check'

export interface WithdrawalAddressViewProps {
  data: WithdrawalAddress
}

export const WithdrawalAddressView = (props: WithdrawalAddressViewProps) => {
  const { data } = props

  useSetPageTitle(data.network.name)

  return (
    <Grid container justify='center' direction='column'>
      <Grid container>
        <Grid item xs={6}>
          <LabelledValue label='Network Type' value={data.network.name} />
        </Grid>
        <Grid item xs={6}>
          <LabelledValue label='Withdrawal Address' value={data.address} />
        </Grid>
      </Grid>
      <Box py={2} />
      <Grid container>
        <Grid item>
          <CheckIcon color='primary' />
        </Grid>
        <Box px={0.75} />
        <Grid item>
          <Typography>
            I understand and agree that InvestaX will check this address against
            fradulent activities
          </Typography>
        </Grid>
      </Grid>
    </Grid>
  )
}
