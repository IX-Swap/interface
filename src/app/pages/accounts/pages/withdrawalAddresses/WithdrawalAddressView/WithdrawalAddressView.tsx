import React from 'react'
import { Grid, Button, DialogActions } from '@material-ui/core'
import { WADialog } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialog'
import { WADialogTitle } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialogTitle'
import { WADialogContent } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressCreate/WADialog/WADialogContent'
import { WAViewContent } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressView/WAViewContent'
import { WithdrawalAddressesRoute } from 'app/pages/accounts/pages/withdrawalAddresses/router/config'
import { useHistory } from 'react-router-dom'

export const WithdrawalAddressView = () => {
  const { replace } = useHistory()

  return (
    <WADialog open>
      <WADialogTitle label='View Blockchain Address' />
      <WADialogContent>
        <WAViewContent />
      </WADialogContent>
      <DialogActions>
        <Grid container xs={12} justifyContent='center'>
          <Button
            variant='contained'
            color='primary'
            disableElevation
            onClick={() => replace(WithdrawalAddressesRoute.list)}
          >
            Ok
          </Button>
        </Grid>
      </DialogActions>
    </WADialog>
  )
}
