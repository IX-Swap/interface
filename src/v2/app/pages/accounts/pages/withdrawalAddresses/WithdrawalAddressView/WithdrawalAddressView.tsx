import React from 'react'
import { Grid, Button } from '@material-ui/core'
import { history } from 'v2/history'
import { WADialog } from 'v2/app/pages/accounts/pages/withdrawalAddresses/WADialog/WADialog'
import { WADialogTitle } from 'v2/app/pages/accounts/pages/withdrawalAddresses/WADialog/WADialogTitle'
import { WADialogContent } from 'v2/app/pages/accounts/pages/withdrawalAddresses/WADialog/WADialogContent'
import { WADialogActions } from 'v2/app/pages/accounts/pages/withdrawalAddresses/WADialog/WADialogActions'
import { WAViewContent } from 'v2/app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressView/WAViewContent'

export const WithdrawalAddressView = ({ isOpen }: { isOpen: boolean }) => {
  return (
    <WADialog open={isOpen}>
      <WADialogTitle label='View Withdrawal Address' />
      <WADialogContent>
        <WAViewContent />
      </WADialogContent>
      <WADialogActions>
        <Grid container xs={12} justify='center'>
          <Button
            variant='contained'
            color='primary'
            disableElevation
            onClick={history.goBack}
          >
            Ok
          </Button>
        </Grid>
      </WADialogActions>
    </WADialog>
  )
}
