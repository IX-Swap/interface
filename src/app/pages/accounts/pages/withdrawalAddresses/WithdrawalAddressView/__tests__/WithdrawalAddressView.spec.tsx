import React from 'react'
import { render } from 'test-utils'
import { WithdrawalAddressView } from 'app/pages/accounts/pages/withdrawalAddresses/WithdrawalAddressView/WithdrawalAddressView'

describe('WithdrawalAddressView', () => {
  it('should match snapshot', () => {
    const { container } = render(<WithdrawalAddressView />)
    expect(container).toMatchSnapshot()
  })
})
