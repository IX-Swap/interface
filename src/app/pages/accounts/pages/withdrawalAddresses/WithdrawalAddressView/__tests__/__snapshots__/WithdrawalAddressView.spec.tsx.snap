// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithdrawalAddressView should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
